### PR TITLE
Kodad water right details map

### PR DESCRIPTION
**Pull request by jeffkodad**
image

image

### CODE REVIEW

File: src/DashboardUI/src/components/Map.tsx [hunk 1]
```diff
@@ -182,6 +182,18 @@ function Map() {
...
   }, [map, circleColors]);
 
+  useEffect(() => {
+    if (!map || mapBounds.length === 0) return;
+    const bounds = new mapboxgl.LngLatBounds(mapBounds[0], mapBounds[0]);
+    mapBounds.forEach(x => {
+      bounds.extend(x);
+    })
+    map.fitBounds(bounds, {
+      padding: 50,
```
Comment (Rhipeus) [inline src/DashboardUI/src/components/Map.tsx:192]: Should we also be able to pass in padding and maxZoom?

**Review by Rhipeus — CHANGES_REQUESTED** [review]

File: src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs [hunk 2]
```diff
@@ -70,5 +70,10 @@ public async Task<List<WaterSourceInfoListItem>> GetWaterRightSourceInfoList(lon
         {
             return await _waterAllocationAccessor.GetWaterRightSourceInfoById(waterRightsId);
         }
+
+        public async Task<List<SiteLocation>> GetWaterRightSiteLocations(long waterRightsId)
```
[the remark this answers is not in the document]
Comment (jeffkodad) [reply src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs]: Updated.

File: src/DashboardUI/src/components/WaterRightMap.tsx [hunk 3]
```diff
@@ -1,8 +1,63 @@
...
+import { useWaterRightSiteLocations } from '../hooks/waterAllocation';
+import GeoJSON from 'geojson';
+import mapboxgl from 'mapbox-gl';
+
+interface waterRightMapProps {
+  waterRightId: string;
+}
+
+function WaterRightMap(props: waterRightMapProps){
+  const waterRightSiteLocations = useWaterRightSiteLocations(+props.waterRightId).data;
```
[the remark this answers is not in the document]
Comment (jeffkodad) [reply src/DashboardUI/src/components/WaterRightMap.tsx]: Added.

File: src/DashboardUI/src/components/Map.tsx [hunk 4]
```diff
@@ -182,6 +182,18 @@ function Map() {
     }
   }, [map, circleColors]);
 
+  useEffect(() => {
+    if (!map || mapBounds.length === 0) return;
+    const bounds = new mapboxgl.LngLatBounds(mapBounds[0], mapBounds[0]);
```
[the remark this answers is not in the document]
Comment (jeffkodad) [reply src/DashboardUI/src/components/Map.tsx]: Yup, the bounds object wants the two opposing corners for coordinates so we give it the same point first before extending to each additional point. This way we don't have to order the points or figure out the corners on our own.

File: src/DashboardUI/src/components/Map.tsx (hunk 1 above)
Comment (jeffkodad) [reply src/DashboardUI/src/components/Map.tsx]: Updated to include those two props.

File: src/DashboardUI/src/config/maps.ts [hunk 5]
```diff
@@ -34,6 +42,36 @@ const mapsJson = {
...
+      "id": "site-locations",
+      "friendlyName": "SiteLocations",
+      "source": "site-locations",
+      "layout": {
+        "visibility": "visible"
+      },
+      "type": "circle",
+      "paint": {
+        "circle-radius": pointCircleRadius,
+        "circle-color": "#01579b"
```
[the remark this answers is not in the document]
Comment (jeffkodad) [reply src/DashboardUI/src/config/maps.ts]: Updated

**Review by Rhipeus — APPROVED** [review]

**Pull request merged**